### ISSUE TITLE
iOS version isn't showing "Today"?

### ISSUE THREAD

**Issue by ilyausorov** [henninghall/react-native-date-picker#72]
First off, this repo is great and much needed to allow us to avoid the annoying modal-based paradigm of the Android date / time pickers that currently exist in React Native. So thanks a ton!

I was able to implement it without any issues, but I did notice that for some reason, the iOS side didn't change the current date to "Today" like it did on the Android side. I believe the DatePickerIOS also does this out of the box. Did I mess something up or did I find an issue?

For reference, here are two screenshots of a) what props I'm entering and b) what I'm seeing (left is Android, right is iOS). Thanks for any suggestions!

![Screen Shot 2019-06-15 at 7 16 01 PM](https://user-images.githubusercontent.com/16529964/59557201-06055f00-8fa2-11e9-8f03-7a13c1703d86.png)
![Screen Shot 2019-06-15 at 7 12 46 PM](https://user-images.githubusercontent.com/16529964/59557191-c3dc1d80-8fa1-11e9-8325-fa31261b9a22.png)

**Comment by henninghall** [comment]
Hi, thanks for you comment! 

Yes I am aware of this bug and it happens on when you change the text color on ios. 

The solution I found for changing the text color of the ios native component "UIDatePicker" had this ugly side effect, but I haven't looked much into if there is a better way. 

If you happen to have time to look into this a bit, it would be really helpful! The code that handles this is found in `ios/RNDatePicker/DatePicker.m` from line 65

**Comment by ilyausorov** [comment]
hey @henninghall so I took a look and it seems that the whole section from lines 67 to 72 in `ios/RNDatePicker/DatePicker.m` is what is causing the current date to not show as "Today". As you can see, we're literally setting the selector`@setHighlightsToday` to `NO`.

`SEL selector = NSSelectorFromString(@"setHighlightsToday:");
NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:[UIDatePicker instanceMethodSignatureForSelector:selector]];
BOOL yes = YES;
[invocation setSelector:selector];
[invocation setArgument:&yes atIndex:2];
[invocation invokeWithTarget:self];`

I removed those lines from the pod, re-installed it into my RN app and re-ran it and indeed the current date now shows up as "Today". 

<img width="346" alt="Screen Shot 2019-06-17 at 1 36 59 PM" src="https://user-images.githubusercontent.com/16529964/59624825-cfe0f000-9105-11e9-8aed-36ef9df3104e.png">

Now this got me thinking that perhaps we can provide the user the option to choose whether they want the current date to show as "Today" or not, by accepting a boolean prop, such as `showCurrentDateAsToday`, reading it the same way that we read the textColor prop, and then using it to determine whether line 70 is YES or NO. What do you think? Alternatively, we can remove those lines and just make both the iOS and Android equivalently show the current date as "Today. Happy to submit a PR with whichever option you prefer.

**Comment by henninghall** [comment]
That's awesome! 

Since I remembered there was something fishy with this I tried to remove those lines myself but get the following result:
<img width="249" alt="Screenshot 2019-06-17 at 20 26 54" src="https://user-images.githubusercontent.com/1652607/59630908-6af5bc00-913e-11e9-86c7-90334d4de93b.png">

Did you have to do anything extra to change the Today text color? If not it might have something to do with the versions, I am using the example project in this repo with versions:
```
simulator: iPhone 5s
ios: 12.2
react-native: 0.56.0
```

Regarding the question regarding the `showCurrentDateAsToday` prop, I think it sounds like a great idea to include it as an optional prop which defaults to true.

**Comment by ilyausorov** [comment]
Wow, that's interesting that you see that. I wonder why! I definitely didn't do anything extra beyond modifying that bit of native code.

I tested with the example repo you provided, and the same simulator (iPhone 5s), iOS (12.2) and react-native (0.56.0) versions, and I still got the same result as above: the today was in the same color as the rest of the dates.

I am also using:
```
Xcode: 10.2.1 (latest)
macOS: 10.14.5 (latest)
react-native-cli: 2.0.1
```

I made this [video](https://drive.google.com/file/d/11WXtGNrDPmUglPG5oZ3EHopqYjRHt3d3/view?usp=sharing) to show my process. What do you think?

**Comment by henninghall** [comment]
Found it! It works fine when you set the text color initially but if you change the prop during runtime you will get my result. It might actually not be that uncommon use case if you for instance have a day and night mode for your app. 

That would be awesome if you found a solution to this, it feels like it should be possible! But if not it might also work to make a hacky workaround where the today string is removed if the color set a second time/during runtime. It would not be a very clear solution but I would still say its definitely better than today.

**Comment by ilyausorov** [comment]
Ahh so that's what it is! Do you think that maybe the way setTextColor is implemented it's only called during the initial onLoad method and then not again?

I can play with it, but I don't have too much Obj C experience so not sure how far I'll get. From what I've also read, this approach that we're already using is utilizing private APIs as Apple's official stance is that the UIDatePicker appearance is not customizable ([link](https://developer.apple.com/documentation/uikit/uidatepicker#2281468))

**Comment by henninghall** [comment]
Sorry for the slow answer!
> Do you think that maybe the way setTextColor is implemented it's only called during the initial onLoad method and then not again? 

It doesn't seem so to me, at least not in our layer if that's what you mean. `setTextColorProp` in `DatePicker.m` will be called every time the prop changes. 

> From what I've also read, this approach that we're already using is utilizing private APIs as Apple's official stance is that the UIDatePicker appearance is not customizable (link)

Yes, I am aware of that this is a gray area but I am not sure it actually using "private API:s" which could get your app rejected. I have never heard anyone using this picker having issues with that. I am not sure of the exact definition but according to [this answer ](https://stackoverflow.com/a/31775498/3522683) this solution does not invoke private APIs. But yes, if you want to be absolutely sure not to break their guidlines you shouldn't change the color.

**Comment by henninghall** [comment]
🚀Released in v2.5.0
